Adding stencil redux docs

corysmc opened the pull request:
Stencil redux docs with a demo walkthrough

bitflower commented on src/docs/guides/redux.md:
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
+
+  async componentWillLoad() {
+    this.store.setStore(configureStore({}));
+    this.store.mapStateToProps(this, (state: MyAppState) => {
```
We should make this:
```
this.storeUnsubscribe = this.store.mapStateToProps(this, (state: MyAppState) => {
```

bitflower commented on src/docs/guides/redux.md:
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
+// /src/components/app-root/app-root.tsx
+import "@stencil/redux";
+import { Component, Prop, State, h } from "@stencil/core";
+import { Store } from "@stencil/redux";
```
Add `Unsubscribe` to imports.

bitflower commented on src/docs/guides/redux.md:
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
+
+  @Prop({ context: "store" })
+  store: Store;
+
```
Add `storeUnsubscribe: Unsubscribe;`

bitflower commented on src/docs/guides/redux.md:
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
+      };
+    });
+  }
+
```
Add:
```
  protected componentDidUnload() {
    if (this.storeUnsubscribe) {
      this.storeUnsubscribe();
    }
  }
```

bitflower commented on src/docs/guides/redux.md:
```diff
@@ -5,8 +5,329 @@ url: /docs/stencil-redux
   - jthoms1
   - adamdbradley
+  - corysmc
 ---
```
If you like my changes add `bitflower` ;-)

corysmc commented:
This is awesome, thanks @bitflower  I seriously just ran into this issue, had no idea what was causing it. Just updated the docs, and added you as a contributor

adamdbradley commented:
🚀
